src/ppx/application.cpp

```diff
@@ -649,10 +649,13 @@ void Application::DispatchInitKnobs()
     InitKnobs();
 }
 
-void Application::DispatchConfig()
+void Application::DispatchConfigAndInitKnobs()
 {
     Config(mSettings);
 
+    // Knobs need to be set up before commandline parsing.
+    DispatchInitKnobs();
```

Comment:
`mKnobManager.UpdateFromFlags(options);` now runs after the first usage of a knob value.

Reply:
The change of initialization here is confusing.
`pAssetsPaths` sounds like a separate PR.

Reply:
Thanks for pointing this out, it is indeed an issue. I did a cleanup of the DispatchConfig() so that it is only doing dispatching config instead of doing more than the name indicated to avoid confusion. And moved all the asset dir related code to a separate function with some comments so that it is clear what are the dependencies.